### PR TITLE
Java Docs

### LINKED ISSUE #48: Add javadocs to public classes/interfaces

### PR DESCRIPTION
**Pull request by VisheshVadhera**
Closes #48 

Added docs for the following classes:

- CustomTypeAdapter
- Logger
- ResponseCacheStore

Also I deleted the Manifest file from rx2-support module since it's no longer an android module.

### CODE REVIEW

File: apollo-runtime/src/main/java/com/apollographql/apollo/CustomTypeAdapter.java [hunk 1]
```diff
@@ -1,8 +1,70 @@
 package com.apollographql.apollo;
 
+/**
+ * CustomTypeAdapter class represents the adapter for converting custom scalar types to Java objects.
```
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/CustomTypeAdapter.java]: `...for converting GraphQL custom scalar types...`

File: apollo-runtime/src/main/java/com/apollographql/apollo/CustomTypeAdapter.java [hunk 2]
```diff
@@ -1,8 +1,70 @@
...
+ *                }
+ *          };
+ * }
+ * </pre>
+ * To use a custom type adapter with Apollo, you must first define the mapping in the build.gradle file:
+ * <pre>   {@code
+ *  apollo {
+ *        generateClasses = true
+ *        customTypeMapping {
+ *        DateTime = "java.util.Date"
```
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/CustomTypeAdapter.java]: We should mention that sometimes when name of the type is reserved keyword in groovy you can use Map notation, for example:
```
customTypeMapping['URL'] = "String"
```
where URL is reserved keyword in groovy and you can't use it (will get compilation error):
```
URL = "String"
```

File: apollo-runtime/src/main/java/com/apollographql/apollo/CustomTypeAdapter.java [hunk 3]
```diff
@@ -1,8 +1,70 @@
...
+ *                ApolloClient client = builder.client;
+ * }
+ * </pre>
+ *
+ * In the example code above, the CustomType.DATE is the class generated by the compiler.
+ */
 public interface CustomTypeAdapter<T> {
 
+  /**
+   * De-serializes the value to the custom scalar type.
```
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/CustomTypeAdapter.java]: We can add: "usually used in parsing GraphQL response"

File: apollo-runtime/src/main/java/com/apollographql/apollo/CustomTypeAdapter.java [hunk 4]
```diff
@@ -1,8 +1,70 @@
...
+   *
+   * @param value the value to de-serialize
+   * @return custom scalar type
+   */
   T decode(String value);
 
+  /**
+   * Serializes the custom scalar type to the corresponding string value.
```
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/CustomTypeAdapter.java]: We can add: `usually used in serializing variables or input values`

File: apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java [hunk 5]
```diff
@@ -4,12 +4,35 @@
 
 import javax.annotation.Nonnull;
 
+/**
+ * ResponseCacheStore is an abstraction for a cache store that is used to read, modify or delete http responses.
```
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java]: Omit "http". This abstraction is used for normalized cache (not http)

File: apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java [hunk 6]
```diff
@@ -4,12 +4,35 @@
...
+ */
 public interface ResponseCacheStore {
+  /**
+   * Returns ResponseCacheRecord for the entry named cacheKey or null if it doesn't exist is not currently readable.
+   *
+   * @param cacheKey the name of the entry
+   * @return ResponseCacheRecord
+   */
   ResponseCacheRecord cacheRecord(@Nonnull String cacheKey) throws IOException;
```
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java]: Could you pls add
`@Nullable ResponseCacheRecord cacheRecord(@Nonnull String cacheKey) throws IOException;`
right @BenSchwab

File: apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java (hunk 5 above)
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java]: Pls ignore my bad

File: apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java [hunk 7]
```diff
@@ -4,12 +4,35 @@
 
 import javax.annotation.Nonnull;
 
+/**
+ * ResponseCacheStore is an abstraction for a cache store that is used to read, modify or delete http responses.
+ */
 public interface ResponseCacheStore {
+  /**
+   * Returns ResponseCacheRecord for the entry named cacheKey or null if it doesn't exist is not currently readable.
```
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java]: Smth wrong here: 
`or null if it doesn't exist is not currently readable`

File: apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java [hunk 8]
```diff
@@ -4,12 +4,35 @@
...
+   */
   ResponseCacheRecord cacheRecord(@Nonnull String cacheKey) throws IOException;
 
+  /**
+   * Returns an editor for the entry named cacheKey or null if another edit is in progress.
+   *
+   * @param cacheKey the entry to edit.
+   * @return {@link ResponseCacheRecordEditor} to use fo editing the entry
+   */
   ResponseCacheRecordEditor cacheRecordEditor(@Nonnull String cacheKey) throws IOException;
```
Comment (sav007) [inline apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java]: pls add  `@Nullable ResponseCacheRecordEditor`

**Comment by VisheshVadhera** [comment]
Made the set of changes suggested. Can be merged if everything else looks fine.

File: apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java [hunk 9]
```diff
@@ -3,13 +3,37 @@
...
+   * @return ResponseCacheRecord
+   */
+  @Nullable ResponseCacheRecord cacheRecord(@Nonnull String cacheKey) throws IOException;
 
-  ResponseCacheRecordEditor cacheRecordEditor(@Nonnull String cacheKey) throws IOException;
+  /**
+   * Returns an editor for the entry named cacheKey or null if another edit is in progress.
+   *
+   * @param cacheKey the entry to edit.
+   * @return {@link ResponseCacheRecordEditor} to use fo editing the entry
```
Comment (marwanad) [inline apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java]: type in `fo`

Comment (VisheshVadhera) [inline apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/ResponseCacheStore.java]: Gosh!

**Comment by sav007** [comment]
@VisheshVadhera merging?

**Comment by VisheshVadhera** [comment]
Yep, have made the requested set of changes, so we can merge.

**Pull request merged**